lib/Extractor/KLEEBuilder.cpp

```diff
@@ -188,11 +188,12 @@ ref<Expr> ExprBuilder::subnuwUB(Inst *I) {
 
 ref<Expr> ExprBuilder::mulnswUB(Inst *I) {
    const std::vector<Inst *> &Ops = I->orderedOps();
-   ref<Expr> L = SExtExpr::create(get(Ops[0]), 2*I->Width);
-   ref<Expr> R = SExtExpr::create(get(Ops[1]), 2*I->Width);
+   Expr::Width Width = get(Ops[0])->getWidth();
```

Comment:
I found I->Width was not giving the correct values in one of the test
cases. I need to find the test case though. Then, I followed the
traditional approach of constructing the expression and find width from it
using getWidth(), it always gave correct values.

On Thu, Feb 26, 2015 at 10:49 AM, Raimondas Sasnauskas <
notifications@github.com> wrote:

> In lib/Extractor/KLEEBuilder.cpp
> <https://github.com/google/souper/pull/179#discussion_r25447766>:
>
> > @@ -188,11 +188,12 @@ ref<Expr> ExprBuilder::subnuwUB(Inst *I) {
> >
> >  ref<Expr> ExprBuilder::mulnswUB(Inst *I) {
> >     const std::vector<Inst *> &Ops = I->orderedOps();
> > -   ref<Expr> L = SExtExpr::create(get(Ops[0]), 2*I->Width);
> > -   ref<Expr> R = SExtExpr::create(get(Ops[1]), 2*I->Width);
> > +   Expr::Width Width = get(Ops[0])->getWidth();
>
> Is there any reason why you are building an expression first to get the
> width instead of writing
> unsigned Width = I->Width;
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/souper/pull/179/files#r25447766>.
>



-- 
Thanks and Regards,
Jubi Taneja